lib/src/util/vulnerabilitySeverityRatings.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+/**
+ * All Possible Vulnerability (CVSS) Severity Ratings.
+ */
+export const allVulnerabilitySeverityRatings = Object.freeze([
+  'none',
+  'low',
+  'medium',
+  'high',
+  'critical'
+] as const);
+
+type SingleDigit = '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9';
+
+export type VulnerabilityScore = `${SingleDigit}.${SingleDigit}` | '10.0';
+
+/**
+ * The type containing only the possible vulnerability CVSS severity rating values.
+ */
+export type VulnerabilitySeverityRating = (typeof allVulnerabilitySeverityRatings)[number];
+
+/**
+ * Lists the ranges of Vulnerability (CVSS) scores and their corresponding severity rating.
+ */
+export const vulnerabilitySeverityAndScoreRangeList = Object.freeze([
+  { min: 9.0, max: 10.0, severity: 'critical' },
+  { min: 7.0, max: 8.9, severity: 'high' },
+  { min: 4.0, max: 6.9, severity: 'medium' },
+  { min: 0.1, max: 3.9, severity: 'low' },
+  { min: 0.0, max: 0.0, severity: 'none' }
+] as const);
+
+const validateVulnerabilityScoreNumber = (scoreNumber: number) => {
+  if (!Number.isInteger(scoreNumber * 10)) {
+    throw new Error('Vulnerability score must be a valid number with one decimal place.');
+  }
+  if (scoreNumber < 0 || scoreNumber > 10) {
+    throw new Error('Vulnerability score must be between 0.0 and 10.0.');
```

Comment:
I typically think `TypeError` is more appropriate for validation errors

Reply:
> I typically think `TypeError` is more appropriate for validation errors

Changed it to TypeError as part of: [9642cc5](https://github.com/sonatype/sonatype-react-shared-components/pull/1202/commits/9642cc5fe1ab3e31c3f982313d993970c398730c)